stdlib/public/runtime/MetadataImpl.h

```diff
@@ -188,6 +189,7 @@ template <class Impl, class T> struct RetainableBoxBase {
   static constexpr size_t stride = sizeof(T);
   static constexpr bool isPOD = false;
   static constexpr bool isBitwiseTakable = true;
+  static const bool isAtomic;
```

Comment:
This will impose a cost for normal atomic code. Can you instead either use a cpp flag or us a constexpr that is set by a cpp flag. In the later case, the optimizer should be able to completely eliminate the false path but limit the amount of places the preprocessor code will touch. Your choice.